test/e2e/storage/volume_metrics.go

```diff
@@ -51,13 +56,17 @@ var _ = SIGDescribe("[Serial] Volume metrics", func() {
 
 		pvc = newClaim(test, ns, "default")
 		var err error
-		metricsGrabber, err = metrics.NewMetricsGrabber(c, nil, false, false, true, false, false)
+		metricsGrabber, err = metrics.NewMetricsGrabber(c, nil, true, false, true, false, false)
 
 		if err != nil {
 			framework.Failf("Error creating metrics grabber : %v", err)
 		}
 	})
 
+	AfterEach(func() {
+		framework.DeletePersistentVolumeClaim(c, pvc.Name, pvc.Namespace)
+	})
```

Comment:
I don't like this very much. This change will be fine, if we were creating pvc in `BeforeEach` but we aren't. We are only instantiating a PVC object in `BeforeEach`. So `AfterEach` is trying to delete a resource that it did not create.  Other places where we delete pvcs in `AfterEach`  - we are also creating them  in `BeforeEach`. 

Moreover, I would like to add a test for `volume_delete` metric which I forgot to cover and hence I will have to move this statement out of `AfterEach` anyways.

Reply:
I did think about this when refactoring - the helper ignores "not found" errors so it should address both the concerns above i.e. handle the case where we didn't end up creating as well as cases where the test needed to explicitly delete the pvc

Reply:
Agree with both of you. @gnufied 's point about symmetry in BeforeEach'es and AfterEach'es is good. @vkamra is correct about the helper.